lib/validateDeps.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# Copyright Contributors to the Open Cluster Management project
+
+# Validate bash version is 4 or greater
+if [ "${BASH_VERSINFO:-0}" -lt 4 ]
+then
+  echo "DEPENDENCY NOT MET. bash version 4 or greater is required to run this tool. Found version ${BASH_VERSION}"
+  echo "On MacOS install with \"brew install bash\".  This bash must be first in your path, but need not be '/bin/bash' or your default login shell."
+  exit 1
```

Comment:
It might be worth trying to include this code directly in `common.sh` (fairly early on before we run into something that requires bash 4) so you can access the common logging functions. That would make it easier to:
- Make sure all error messages go to stderr
- Prefix errors with `error: ` as is done everywhere else in `ck` (see `error`, `fatal`, `verbose` functions... I would probably use `error` for the main error message, and then `verbose 0` for the text explaining how to recover)